src/Multi.php

```diff
@@ -52,6 +54,28 @@ public static function zipLongest(iterable ...$iterables): \MultipleIterator
         return $zippedIterator;
     }
 
+    /**
+     * Iterate multiple iterable collections simultaneously.
+     *
+     * Works like Multi::zipStrict() method
+     * but throws OutOfRangeException if at least one iterator ends before the others.
+     *
+     * @param iterable<mixed> ...$iterables
+     *
+     * @return \MultipleIterator<mixed>
+     * @throws \OutOfRangeException
+     */
+    public static function zipStrict(iterable ...$iterables): \MultipleIterator
```

Comment:
Suggestion: How about naming this `zipEqual`? The term "strict" in PHP generally refers to types adhering to declarations and not allowing weak type rules from applying. I don't think "strict" implies that the lengths need to be equal. "Equal," on the other hand, provides that information.

Reply:
Done.